### ISSUE TITLE
WinForms - OnHandleDestroyedInternal may crash a process when ChromiumWebBrowser is disposed

### ISSUE THREAD

**Issue by jozefizso** [cefsharp/CefSharp#3252]
- **What version of the product are you using?**
    - CefSharp.Winforms (v84.4.10) nuget package

- **What architecture x86 or x64?**
    - x86
    - x64
    
- **On what operating system?**
    - Windows 10

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
    - WinForms
    
- **What steps will reproduce the problem?**
    - Create add-in for Microsoft Office using .NET Framework 4.8
    - Host the `ChromiumWebBrowser` in a Task Pane inside a `UserControl`
    - When document in MS Office is closed, dispose the `UserControl` with the `ChromiumWebBrowser`
    - `ChromiumWebBrowser.OnHandleDestroyedInternal()` may crash the dispose code with `FileNotFoundException`

- **What is the expected output? What do you see instead?**

Exception we see when we dispose the `ChromiumWebBrowser`:

```
System.IO.FileNotFoundException: Could not load file or assembly 'CefSharp.Core, Version=84.4.10.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138' or one of its dependencies. The system cannot find the file specified.
File name: 'CefSharp.Core, Version=84.4.10.0, Culture=neutral, PublicKeyToken=40c4b6fc221f4138'
  at CefSharp.WinForms.ChromiumWebBrowser.OnHandleDestroyedInternal()
  at CefSharp.WinForms.ChromiumWebBrowser.OnHandleDestroyed(EventArgs e)
  at System.Windows.Forms.Control.WmDestroy(Message& m)
  at System.Windows.Forms.Control.WndProc(Message& m)
  at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)
```

- **Please provide any additional information below.**
    - We don't have any detailed logs available yet
    - We hit a similar issue in v83.4.20 release https://github.com/cefsharp/CefSharp/issues/3219

- **Does this problem also occur in the `CEF` Sample Application**
    - No it does not.

**Comment by amaitland** [comment]
What version of Office are you using? 

What sort of Office addin are you creating? VSTO? Have they added something new?

The steps to reproduce are rather vaigue, please add more detail. Review the list from the point of view of someone who has never written an office add-in. 

> may crash the dispose code with 

Please clarify what you mean by may crash. Is the exception always thrown?

I don't own Office, so this isn't something I'll be debugging.

**Comment by jozefizso** [comment]
It was Office 365, but customer did not provide exact version and we are still waiting for logs from the customer to see what happened.

The addin is standard .NET Framework add-in for MS Office without using VSTO.
The addin uses the custom `AppDomain.AssemblyResolve` handler to resolve the correct CefSharp libraries based on 32/64bit version of MS Office.

When user closes the document, we will dispose our UserControl in the task pane which will dispose the `ChromiumWebBrowser`. When we used the CefSharp **83.4.20** this disposal crash the PowerPoint at this point. Also our Visual Studio was crashing when the UserControl was opened in the WinForms Designer and it was closed. This was fixed in #3219.

We updated to use CefSharp **84.4.10** and Visual Studio is working correctly and PowerPoint does not crash on our dev machines nor on QA machines.

But we got report from one customer with this issue that **CefSharp** is crashing in `ChromiumWebBrowser.OnHandleDestroyedInternal` method.

---

I am not sure if this is because of problem on customer's PC because the `CefSharp.Core.dll` cannot be loaded for some unknown reasons (eg #3246) or there is still an issue in

**Comment by amaitland** [comment]
> It was Office 365, but customer did not provide exact version and we are still waiting for logs from the customer to see what happened.

Perhaps a bit too soon to open a bug here wouldn't you say? You haven't finished collecting data.

> The addin is standard .NET Framework add-in for MS Office without using VSTO

I'm not familiar with this sort of add-in please provide a link with details. Does it use Appdomains in any way?

> Also our Visual Studio was crashing when the UserControl was opened in the WinForms Designer and it was closed. This was fixed in #3219.

The WinForms Designer is a very specific case where we deliberately avoid loading CefSharp.Core.dll if it cannot be loaded at runtime then that's a problem you need to resolve,  nothing CefSharp has control over.

> I am not sure if this is because of problem on customer's PC because the CefSharp.Core.dll cannot be loaded for some unknown reasons (eg #3246) or there is still an issue in

What issue would that be exactly? What makes you think there's a bug in CefSharp?
